bot/cogs/snakes.py

```diff
@@ -15,31 +38,67 @@ class Snakes:
     def __init__(self, bot: AutoShardedBot):
         self.bot = bot
 
+    @staticmethod
+    def snake_url(name) -> str:
+        """Get the URL of a snake"""
+
+        def encode_url(text):
+            """Encode a string to URL-friendly format"""
+            return BASEURL.format(text.replace(' ', '%20').replace("'", '%27'))
+
+        # Check if the snake name is known
+        if name.upper() in list(map(lambda n: n.upper(), SNAKES)):
+            return encode_url(name)
+
+        # Get a list of similar names if a match wasn't found
+        matches = difflib.get_close_matches(name, SNAKES, n=1)
+        return encode_url(matches[0] if matches != [] else random.choice(SNAKES))
+
+    @staticmethod
+    async def fetch(session, url):
+        """Fetch the contents of a URL as a json"""
+        async with async_timeout.timeout(10):
+            async with session.get(url) as response:
+                return await response.json()
+
     async def get_snek(self, name: str = None) -> Dict[str, Any]:
-        """
-        Go online and fetch information about a snake
+        """Get a snake with a given name, or otherwise randomly"""
+        if name is None:
+            name = random.choice([x.strip('\n') for x in SNAKES])
+            
+        elif name.upper() == "PYTHON":
+            return PYTHON
 
-        The information includes the name of the snake, a picture of the snake, and various other pieces of info.
-        What information you get for the snake is up to you. Be creative!
+        # Get snake information
+        async with aiohttp.ClientSession() as session:
+            url = self.snake_url(name)
 
-        If "python" is given as the snake name, you should return information about the programming language, but with
-        all the information you'd provide for a real snake. Try to have some fun with this!
+            # Get the 
+            response = await self.fetch(session, url)
+            page = response["query"]["pages"]
+            content = next(iter(page.values()))
 
-        :param name: Optional, the name of the snake to get information for - omit for a random snake
-        :return: A dict containing information on a snake
-        """
+            # Parse the full-res image from the thumbnail
+            thumb = content.get("thumbnail", {}).get("source", "http://i.imgur.com/HtIPyLy.png/beep")
+            image = "/".join(thumb.replace("thumb/", "").split("/")[:-1])
+            
+            # WHY WOULD YOU USE DICTIONARY LITERAL IN A RETURN STATEMENT but okay lol
```

Comment:
this seems fine from the way you've done this - not sure I have any qualms with it but maybe consider making a class for this return object...